plugins/woocommerce/includes/rest-api/Controllers/Version2/class-wc-rest-system-status-v2-controller.php

```diff
@@ -32,6 +32,30 @@ class WC_REST_System_Status_V2_Controller extends WC_REST_Controller {
 	 */
 	protected $rest_base = 'system_status';
 
+	/**
+	 * Register cache cleaner
+	 *
+	 * Handles all the cache cleaning for this endpoint. We need to register
+	 * these functions before the routes are registered, so this function gets
+	 * called from Server.php
+	 */
+	public static function register_cache_clean() {
+		// Clear the theme cache if we switch themes or our theme is upgraded.
+		add_action( 'switch_theme', array( __CLASS__, 'clean_theme_cache' ) );
+		add_action(
+			'upgrader_process_complete',
+			function( $upgrader, $extra ) {
+				if ( 'theme' !== $extra['type'] ) {
```

Comment:
The cached theme data has a couple of props whose values look like they depend on the version of WC (thinking of `has_outdated_templates` and `parent_version_latest`). So it might make sense to clear the cache when WC (and maybe extensions) get updated as well.